includes/wcpdf-functions.php

```diff
@@ -241,4 +241,49 @@ function wcpdf_output_error( $message, $level = 'error', $e = null ) {
  */
 function wcpdf_date_format( $document = null, $date_type = null ) {
 	return apply_filters( 'wpo_wcpdf_date_format', wc_date_format(), $document, $date_type );
+}
+
+/**
+ * Catch MySQL errors
+ * Original from here: https://github.com/johnbillion/query-monitor/blob/d5b622b91f18552e7105e62fa84d3102b08975a4/collectors/db_queries.php#L125-L280
+ *
+ * @param  object $wpdb
+ * @return array  Errors found
+ */
+function wcpdf_catch_db_object_errors( $wpdb ) {
+	global $EZSQL_ERROR;
+
+	$errors = array();
+
+	// with SAVEQUERIES defined as false, `wpdb::queries` is empty but `wpdb::num_queries` is not.
+	if ( empty( $wpdb ) || empty( $wpdb->queries ) ) {
```

Comment:
if you're bailing when `$wpdb->queries` is empty, you never reach the `$EZSQL_ERROR` fallback. Perhaps you should reverse it, by including the loop inside a `! empty( $wpdb ) && ! empty( $wpdb->queries )`, then check the fallback after that?